fido/fido.py

```diff
@@ -200,18 +221,23 @@ def fetch_inner(
     reactor = _import_reactor()
 
     agent = get_agent(reactor, connect_timeout, tcp_nodelay)
+    new_headers = listify_headers(twisted_headers)
+
+    if gzip_enabled:
+        if 'gzip' not in new_headers.getRawHeaders('accept-encoding', []):
```

Comment:
I just realized that we don't check for accepted encodings with quality values (https://developer.mozilla.org/en-US/docs/Glossary/Quality_Values). If 'gzip;q=1.0' is included in the accept-encoding headers, then we'll be adding a duplicate 'gzip' header to the list. Is this something we should support?

Reply:
I didn't know about this, but if it's in the spec we could check for the gzip substring in the header value. Good call!

Reply:
it looks like this is correct!